File: metodos/reglaFalsa.py
def regla_falsa(f, a, b, tol, max_iter):
    i = 0
    try:
        fa = eval(f.replace('x', 'a'))
        fb = eval(f.replace('x', 'b'))
        while abs(b - a) > tol and i < max_iter:
            c = (a * fb - b * fa) / (fb - fa)
            fc = eval(f.replace('x', 'c'))
            if fa * fc < 0:
                b = c
                fb = fc
            else:
                a = c
                fa = fc
            i += 1
        if abs(b - a) > tol:
            print("El método no converge después de %d iteraciones." % max_iter)
            return None
        return "La raiz de la funcion es: " + str(c)
    except ZeroDivisionError:
        print("Error: División por cero.")
        return None



# # intervalo y toleracia
# f = "x**2 -2"
# a = 1
# b = 2
# tol = 1e-6
# max_iter = 100

# raiz = regla_falsa(f, a, b, tol, max_iter)
# print(raiz)

File: metodos/MetodosJava/ReglaFalsa.java
import java.util.function.Function;

public class ReglaFalsa {
    public static String reglaFalsa(Function<Double, Double> f, double a, double b, double tol, int maxIter) {
        int iter = 0;
        double fa = f.apply(a);
        double fb = f.apply(b);

        while (Math.abs(b - a) > tol && iter < maxIter) {
            double c = (a * fb - b * fa) / (fb - fa);
            double fc = f.apply(c);

            if (fa * fc < 0) {
                b = c;
                fb = fc;
            } else {
                a = c;
                fa = fc;
            }

            iter++;
        }

        if (Math.abs(b - a) > tol) {
            return "El método no converge después de " + maxIter + " iteraciones.";
        } else {
            return "La raíz de la función es: " + c;
        }
    }
}
